### ISSUE TITLE
Crash on reset default runtime optimization

### ISSUE THREAD

**Issue by shuvashish76** [MuntashirAkon/AppManager#1131]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

"Reset to default" runtime optimization crashes the app.

### To Reproduce

1. Open "1-Click Ops"
2. Click on "Perfrom runtime optimization > Reset to default"
3. See error

### Logs

```
java.lang.IllegalArgumentException: reason -1 invalid
    at android.os.Parcel.createExceptionOrNull(Parcel.java:2430)
    at android.os.Parcel.createException(Parcel.java:2410)
    at android.os.Parcel.readException(Parcel.java:2393)
    at android.os.Parcel.readException(Parcel.java:2335)
    at android.content.pm.IPackageManager$Stub$Proxy.forceDexOpt(IPackageManager.java:8133)
    at io.github.muntashirakon.AppManager.apk.behavior.DexOptimizer.forceDexOpt(DexOptimizer.java:84)
    at io.github.muntashirakon.AppManager.batchops.BatchOpsManager.opPerformDexOpt(BatchOpsManager.java:844)
    at io.github.muntashirakon.AppManager.batchops.BatchOpsManager.performOp(BatchOpsManager.java:287)
    at io.github.muntashirakon.AppManager.batchops.BatchOpsManager.performOp(BatchOpsManager.java:227)
    at io.github.muntashirakon.AppManager.batchops.BatchOpsService.onHandleIntent(BatchOpsService.java:157)
    at io.github.muntashirakon.AppManager.types.ForegroundService$ServiceHandler.handleMessage(ForegroundService.java:54)
    at android.os.Handler.dispatchMessage(Handler.java:111)
    at android.os.Looper.loopOnce(Looper.java:238)
    at android.os.Looper.loop(Looper.java:357)
    at android.os.HandlerThread.run(HandlerThread.java:67)
```

### Device info

App version: 4.0.0-alpha01
App version code: 434
Android build version: 8049b-fa5cf
Android release version: 12
Android SDK version: 31
Android build ID: S2RUBS32.51-15-9-7
Device brand: motorola
Device manufacturer: motorola
Device name: corfur
Device model: moto g71 5G
Device product name: corfur_g
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-IN
In-App Language: auto
Mode: auto
Inferred Mode: Root

**Comment by MuntashirAkon** [comment]
There's a bug in the Android framework. From the [relevant section](https://cs.android.com/android/platform/superproject/+/android-12.0.0_r1:frameworks/base/services/core/java/com/android/server/pm/PackageManagerService.java;l=13149):

```java
        synchronized (mInstallLock) {
            Trace.traceBegin(TRACE_TAG_PACKAGE_MANAGER, "dexopt");

            // Whoever is calling forceDexOpt wants a compiled package.
            // Don't use profiles since that may cause compilation to be skipped.
            final int res = performDexOptInternalWithDependenciesLI(pkg, pkgSetting,
                    new DexoptOptions(packageName,
                            getDefaultCompilerFilter(),
                            DexoptOptions.DEXOPT_FORCE | DexoptOptions.DEXOPT_BOOT_COMPLETE));

            Trace.traceEnd(TRACE_TAG_PACKAGE_MANAGER);
            if (res != PackageDexOptimizer.DEX_OPT_PERFORMED) {
                throw new IllegalStateException("Failed to dexopt: " + res);
            }
        }
```

`DexoptOptions` here did not specify a reason. As a result, `-1` (an invalid reason) was set by default which resulted in a crash as soon as the framework tried to figure out the name of the reason (which must be ≥ 0). A possible workaround is doing this manually in the app.

**Comment by MuntashirAkon** [comment]
For future reference, [this](https://cs.android.com/android/platform/superproject/+/android-12.0.0_r1:frameworks/base/services/core/java/com/android/server/pm/PackageDexOptimizer.java;l=412) is where the crash was happening:

```java
    private String getAugmentedReasonName(int compilationReason, boolean useDexMetadata) {
        String annotation = useDexMetadata
                ? ArtManagerService.DEXOPT_REASON_WITH_DEX_METADATA_ANNOTATION : "";
        return getReasonName(compilationReason) + annotation;
    }
```

where `reasonName(int)` is:

```java
    public static String getReasonName(int reason) {
        if (reason < 0 || reason >= REASON_STRINGS.length) {
            throw new IllegalArgumentException("reason " + reason + " invalid");
        }
        return REASON_STRINGS[reason];
    }
```

Since the compilation reason is always `-1`, performing dex opt in A12 using `PackageManager(Service)#forceDexOpt(String)` will always fail.

**Comment by MuntashirAkon** [comment]
Should be fixed by a82016ffc95ea93f9b2a45ce046fe596c2968798
